### PR TITLE
Increased coverage on Helpers: content.go.

### PR DESCRIPTION
**Pull request by Skarlso**
Also, some minor tweaks to content.go. Removed two unused functions and did a little formatting here and there.

### CODE REVIEW

**Comment by Skarlso** [comment]
Checking now.

File: helpers/content_test.go [hunk 1]
```diff
@@ -117,3 +122,370 @@ func TestTruncateWordsByRune(t *testing.T) {
...
+	headerBuffer := &bytes.Buffer{}
+	footnoteBuffer := &bytes.Buffer{}
+	expectedFootnoteHref := []byte("href=\"#fn:testid:href\"")
+	expectedHeaderID := []byte("<h1 id=\"id\"></h1>")
+
+	actualRenderer.FootnoteRef(footnoteBuffer, []byte("href"), 1)
+	actualRenderer.Header(headerBuffer, func() bool { return true }, 1, "id")
+
+	if !bytes.Contains(footnoteBuffer.Bytes(), expectedFootnoteHref) {
+		t.Errorf("Footnote anchor prefix not applied. Actual:%s Expected:%s", footnoteBuffer.Bytes(), expectedFootnoteHref)
```
Comment (bep) [inline helpers/content_test.go]: I think buff. String() is what the user wants to see printed here ... or use something other thant %s.

File: helpers/content_test.go [hunk 2]
```diff
@@ -117,3 +122,370 @@ func TestTruncateWordsByRune(t *testing.T) {
...
+	}
+	ctx := &RenderingContext{}
+	ctx.Config = ctx.getConfig()
+	ctx.Config.Extensions = []string{"headerId"}
+	ctx.Config.ExtensionsMask = []string{"noIntraEmphasis"}
+
+	actualFlags := getMarkdownExtensions(ctx)
+	if actualFlags&blackfriday.EXTENSION_NO_INTRA_EMPHASIS == blackfriday.EXTENSION_NO_INTRA_EMPHASIS {
+		t.Errorf("Masked out flag {%v} found amongts returned extensions.", blackfriday.EXTENSION_NO_INTRA_EMPHASIS)
+	}
```
Comment (bep) [inline helpers/content_test.go]: It think it is "amongst". See other spelling(s) above.

File: helpers/content_test.go [hunk 3]
```diff
@@ -117,3 +122,370 @@ func TestTruncateWordsByRune(t *testing.T) {
...
+
+	if !bytes.Equal(actualToc, expectedToc) {
+		t.Errorf("Actual toc (%s) did not equal expected (%s) toc content", actualToc, expectedToc)
+	}
+}
+
+func TestRenderBytesWithTOC(t *testing.T) {
+	//Since these tests require an envionment to be set up with various
+	//third party components, I have to disable them for now.
+	t.SkipNow()
```
Comment (bep) [inline helpers/content_test.go]: If you have to disable it ... delete it.

File: helpers/content_test.go [hunk 4]
```diff
@@ -117,3 +122,370 @@ func TestTruncateWordsByRune(t *testing.T) {
...
+		if !bytes.Equal(actualToc, test.expected) {
+			t.Errorf("Actual (%s) did not match expected (%s)", actualToc, test.expected)
+		}
+	}
+}
+
+func TestRenderBytes(t *testing.T) {
+	//Since these tests require an envionment to be set up with various
+	//third party components, I have to disable them for now.
+	t.SkipNow()
```
Comment (bep) [inline helpers/content_test.go]: Ditto.

File: helpers/content_test.go [hunk 5]
```diff
@@ -117,3 +122,370 @@ func TestTruncateWordsByRune(t *testing.T) {
...
+		actualToc := RenderBytes(ctx)
+		if !bytes.Equal(actualToc, test.expected) {
+			t.Errorf("Actual (%s) did not match expected (%s)", actualToc, test.expected)
+		}
+	}
+}
+
+func TestTotalWords(t *testing.T) {
+	t.SkipNow()
+	testString := "Two , Words!"
```
Comment (bep) [inline helpers/content_test.go]: Fix test case, re discussion.

File: helpers/content_test.go [hunk 6]
```diff
@@ -117,3 +122,370 @@ func TestTruncateWordsByRune(t *testing.T) {
...
+	actualWordCount := TotalWords(testString)
+
+	if actualWordCount != 2 {
+		t.Errorf("Actual word count (%d) for test string (%s) did not match 2.", actualWordCount, testString)
+	}
+}
+
+func TestWordCount(t *testing.T) {
+	t.SkipNow()
```
Comment (bep) [inline helpers/content_test.go]: Skip?

**Comment by bep** [comment]
Looks mostly fine ... but the "renderer flag tests" looks a bit verbose to me. I haven't looked closely, but are you sure all those variants are needed to get 100% test coverage in `go gest`? I think they should be weighed against the cost of adding a new Blackfriday flag ... do I have to also update 5 tests cases?

File: helpers/content_test.go (hunk 1 above)
Comment (Skarlso) [inline helpers/content_test.go]: You mean, instead of footnoteBuffer.Bytes(), since I'm displaying %s, should rather be footnoteBuffer.String() ?

**Comment by Skarlso** [comment]
I'll check the render tests. It felt a bit verbose indeed. I do think that setting one, should be fine to get a proper coverage of the statements. Thanks.

**Comment by Skarlso** [comment]
I have to delete this pull and create a new with just one commit. My branch is completely destroyed.

**Comment by spf13** [comment]
@Skarlso Doing a `git pull --rebase` will eliminate all these "Merge remote-tracking branch" lines and keep the branch cleaner. Then doing the rebase squash will be much easier.

**Comment by Skarlso** [comment]
Ahh thanks @spf13. That was what I was looking for! Much appreciated.

**Pull request closed without merging**